packages/nocms-plugin-json-resources/src/createJsonResourceProvider.js

```diff
@@ -3,7 +3,19 @@ import path from 'path';
 const jsonResourceGlobPattern = '**/!(_)*.json';
 const jsonResourceType = 'json';
 
-export default function createJsonResourceProvider ({findFiles, readFile, watchFiles, writeFile}) {
+function canCompileJsonResource (jsonResource) {
+	return jsonResource.type === jsonResourceType;
```

Comment:
Is it possible that `jsonResource` is undefined? --> `can't read type of undefined`

Reply:
I'll throw if either `jsonResource` or `.type` is undefined